add probe-rs-meta (#3984)
probe-rs/probe-rs: probe-rs-meta/README.md

# probe-rs-meta

This crate allows embedding metadata into ELF binaries so that probe-rs can autodetect it.
This way you can run the tests by simply doing `probe-rs run <ELF>`, without adding
any extra flags.

## Usage

You can specify metadata, for example:

```rust

> probe_rs_meta::chip!(b"rpi-pico");

Reviewer: ```suggestion
probe_rs_meta::chip!(b"rp2040");
```

rp-pico is not a valid probe-rs chip identifier

# probe-rs-meta

This crate allows embedding metadata into ELF binaries so that probe-rs can autodetect it.
This way you can run the tests by simply doing `probe-rs run <ELF>`, without adding
any extra flags.

## Usage

First, include the `probe-rs.x` linker script. Either via `build.rs` (recommended)

```rust

> println!("cargo:rustc-link-arg-bins=-Tteleprobe.x");

Reviewer 2: ```suggestion
println!("cargo:rustc-link-arg-bins=-Tprobe-rs.x");
```

arguably it would be good to call this `probe-rs-meta.x` :)

```

or in `.cargo/config.toml` (older way, not recommended)

```toml
[target.'cfg(all(target_arch = "arm", target_os = "none"))']
rustflags = [
  "-C", "link-arg=-Tprobe-rs.x",
]
```

Then, you can specify metadata, for example:

```rust
probe_rs_meta::target!(b"rpi-pico");
```

## Minimum supported Rust version (MSRV)

> `teleprobe-meta` is guaranteed to compile on the latest stable Rust version at the time of release. It might compile with older versions but that may change in any new patch release.

Reviewer 2: ```suggestion
`probe-rs-meta` is guaranteed to compile on the latest stable Rust version at the time of release. It might compile with older versions but that may change in any new patch release.
```
